packages/template-retail-react-app/app/commerce-api/hooks/useBasket.js

```diff
@@ -69,18 +69,8 @@ export default function useBasket(opts = {}) {
                 // We only support single baskets for now. Grab the first one.
                 let basket = Array.isArray(customerBaskets?.baskets) && customerBaskets.baskets[0]
 
-                if (!basket) {
-                    // Back to using ShopperBaskets for all basket interaction.
-                    basket = await api.shopperBaskets.createBasket({})
-
-                    // Throw if there was a problem creating the basket
-                    if (isError(basket)) {
-                        throw new Error(basket)
-                    }
-                }
-
                 // Update basket currency if it was created with the wrong one, this will also set the state.
-                if (currency && basket.currency !== currency) {
+                if (basket && currency && basket.currency !== currency) {
                     await this.updateBasketCurrency(currency, basket.basketId)
                 } else {
                     setBasket(basket)
```

Comment:
On this line, it's possible that `basket` is undefined.. is it ok to pass in `undefined` to `setBasket`?

Reply:
Same as above.
Reverted useBasket back to original code.